src/content/developers/docs/smart-contracts/upgrading-smart-contracts/index.md

```diff
@@ -0,0 +1,163 @@
+
+When a user makes a function call, the proxy contract checks the mapping to find the facet responsible for executing that function. Then it invokes `delegatecall` (using the fallback function) and redirects the call to the appropriate logic contract.  
+
+The diamond upgrade pattern has some advantages over traditional proxy upgrade patterns:
+
+1. It allows you to upgrade a small part of the contract without changing all of the code. Using the proxy pattern for upgrades requires creating an entirely new logic contract, even  for minor upgrades. 
+
+2. All smart contracts (including logic contracts used in proxy patterns) have a 24KB size limit, which can be a limitation—especially for complex contracts requiring more functions. The diamond pattern makes it easy to solve this problem by splitting functions across multiple logic contracts. 
+
+3. Proxy patterns adopt a catch-all approach to access controls. An entity with access to upgrade functions can change the _entire_ contract. But the diamond pattern enables a modular permissions approach, where you can restrict entities to upgrading certain functions within a smart contract. 
+
+[More on the diamond pattern](https://eip2535diamonds.substack.com/p/introduction-to-the-diamond-standard?s=w). 
+
+## Pros and cons of upgrading smart contracts {#pros-and-cons-of-ugrading-smart-contracts}
+
+| Pros                                | Cons                                                                             |
+| ----------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------ |
+| A smart contract upgrade can make it easier to fix vulnerabilities discovered in the post-deployment phase.            | Upgrading smart contracts negates the idea of code immutability, which has implications for decentralization and security.                                          |
+| Developers can use upgrades to add new features to smart contract | Users must trust developers not to modify smart contracts arbitrarily.  |
+| Smart contract upgrades can improve safety for end-users since bugs can be fixed quickly.                              | Programming upgrade functionality into smart contracts adds another layer of complexity and increases the possibility of critical flaws.                        |
+| The opportunity to upgrade smart contracts may encourage developers to launch projects faster without doing due dilligence during the development phase. | Insecure access control or centralization in smart contracts can make it easier for malicious actors to perform unauthorized upgrades.
```

Comment:
> The opportunity to upgrade smart contracts may encourage developers to launch projects faster without doing due dilligence during the development phase.

Is this supposed to be in the "pro" column worded this way?

(Also, "diligence")

Reply:
That's definitely a mistake. I'll move it to the cons section.